CitationStyleLanguagePlugin.php

```diff
@@ -583,15 +583,23 @@ public function getCitation(PKPRequest $request, Submission $submission, string
                     $additionalMarkup = [
                         'DOI' => [
                             'function' => function ($item, $renderedValue) {
-                                return '<a href="https://doi.org/' . $item->DOI . '">' . $renderedValue . '</a>';
+                                $doiWithUrl = 'https://doi.org/'.$item->DOI;
+                                if (str_contains($renderedValue, $doiWithUrl)) {
+                                    $doiLink = '<a href="' . $doiWithUrl . '">' . $doiWithUrl . '</a>';
+                                    return str_replace($doiWithUrl, $doiLink, $renderedValue);
+                                } else {
+                                    $doiLink = '<a href="' . $doiWithUrl . '">' . $item->DOI . '</a>';
+                                    return str_replace($item->DOI, $doiLink, $renderedValue);
+
+                                }
```

Comment:
If there will be always `https://doi.org/{DOI}` or `{DOI}`, I think it makes sense :)

If you want something shorter, this should probably work (just added the `https://doi.org/` as something optional):
```php
preg_replace('#(?:https://doi.org/)?(' . preg_quote($itemDOI, '#') . ')#i', "<a href=\"https://doi.org/$1\">$0</a>", $renderedValue);
```

Reply:
I will leave it as it is for now... :-)